redaxo/src/addons/structure/lib/structure_data.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * @package redaxo5
+ */
+
+class rex_structure_data
```

Comment:
Ohne es vollständig durchdacht zu haben: ggf. Wäre es hier sinnvoll die klasse rex_structure_context zu nennen und diese von rex_context erben zu lassen

Reply:
Bei Verwendung von rex_context liefe es offiziell über getParam($name), oder nicht?

Reply:
rex_context impliziert ja dass die params die du in dem context verwendest autom. auch als hidden field mitgegeben werden bzw. in urls angehongen werden.

dies wird vermutlich nicht für alle deine parameter in dieser form mit jeglichen urls passen...?

Reply:
Das halte ich an dieser Stelle für unproblematisch. Aktuell werden ja lediglich die allgemeinen Parameter, die zum Aufruf der Seite gesetzt werden übernommen:
https://github.com/redaxo/redaxo/blob/master/redaxo/src/addons/structure/pages/index.php#L8-L17
Wenn man sie gezielt übergibt und nicht pauschal die kompletten Globals, sollte das für die jeweils aktuelle Struktur-Seite stets den korrekten Kontext bieten. Das Ziel ist ja, den aktuellen Kontext stets abrufbereit für Fragemente etc. vorzuhalten. Dort können sie ja immer noch manipuliert und lokal überschrieben werden.

Reply:
Ansonsten sind in den Methoden lediglich noch die Plausibilitätsprüfungen enthalten. Die halte ich dort auch für sinnvoll.

Reply:
Evtl. müsste man bei der Ausgabe zwischen den Basic-Parametern und denzusätzlichen Parametern unterscheiden.

Reply:
jepp, man müsste mal sehen wie es aussieht. am besten mal prototypisch für einen bereich umsetzen (nicht für alle dateien aus dem diff)

Reply:
Ich habe das Ganze noch mal durchgespielt. Perspektivisch sehe ich auch die SQL-Objekte und Pager im Container, da sie häufig in den Fragments benötigt werden. Stellt sich die Frage ob rex_context da noch der passende Parent ist?

Reply:
jepp sowas dachte ich mir. besser einen `rex_structure_context` ohne parent und dort eine getContext() methode dann rein